src/Framework/Profiler/ProfilerResult.cs

```diff
@@ -73,15 +73,14 @@ public ProfiledLocation(TimeSpan inclusiveTime, TimeSpan exclusiveTime, int numb
         /// <inheritdoc />
         public override bool Equals(object obj)
         {
-            if (!(obj is ProfiledLocation))
+            if (obj is ProfiledLocation location)
```

Comment:
nit:
return obj is ProfiledLocation location &&
           InclusiveTime.Equals(location.InclusiveTime) &&
...
?